cmake-modules/AzureVcpkg.cmake

```diff
@@ -18,13 +18,50 @@ macro(az_vcpkg_integrate)
   endif()
 endmacro()
 
-macro(az_vcpkg_export targetName macroNamePart dllImportExportHeaderPath)
+macro(az_vcpkg_portfile_prep targetName fileName contentToRemove)
+  # with sdk/<lib>/vcpkg/<fileName>
+  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/${fileName}" fileContents)
+
+  # Windows -> Unix line endings
+  string(FIND fileContents "\r\n" crLfPos)
+  string(REPLACE "\r\n" "\n" fileContents ${fileContents})
+
+  # remove comment header
+  string(REPLACE "${contentToRemove}" "" fileContents ${fileContents})
+
+  # undo Windows -> Unix line endings (if applicable)
+  if (crLfPos GREATER -1)
+    string(REPLACE "\n" "\r\n" fileContents ${fileContents})
+  endif()
+  unset(crLfPos)
```

Comment:
Why is this line ending normalizing/reverting necessary?

Reply:
This trimming logic is a bit more complex than I initially expected. I am wondering if its better to just remove the copyright headers from the checked-in file OR maybe leave it as is and ask vcpkg folks for an exception.

This PR already got merged with the copyright headers: https://github.com/microsoft/vcpkg/pull/16183

Reply:
We find a 3-line string and replace it with ann empty one. Without it, find and replace would fail when Windows line endings are used (git checks out with Windows line ending on Windows, Unix endings on Max and Linux).

Reply:
I think it is ok now. In VcPkg, no other package has these headers.
In our repo, we have them for every file.
This logic removes the exact comment, or does nothing. I deliberately not using any smarter than necessary logic here - not using regexes, but not removing "just first 3 lines" either. In case you don't have any copyright header, nothing happens. In case you have something else in the header, nothing happens, it goes to PR. And in case when it is this exact header, it gets removed. I like it.